test/files/pjs003/fail-pjs003-IFC4_3_ADD2_GuidTests.ifc

```diff
@@ -0,0 +1,42 @@
+ISO-10303-21;
+HEADER;
+
+FILE_DESCRIPTION((''),'2;1');
+FILE_NAME('','2024-06-12T09:50:55',(''),(''),'LAB','GMN3DS','');
+FILE_SCHEMA(('IFC4X3_ADD2'));
+ENDSEC;
+
+DATA;
+
+/* PROJECT CONTEXT  */
+#1= IFCPROJECT('2eS$fZmw9QmetNIywDEzSR',#2,'ODT',$,$,$,$,(#10),#15);
+#2= IFCOWNERHISTORY(#3,#6,$,.NOCHANGE.,$,$,$,152828);
+#3= IFCPERSONANDORGANIZATION(#4,#5,$);
+#4= IFCPERSON($,'','ALCHEMIST',$,$,$,$,$);
+#5= IFCORGANIZATION($,'LAB',$,$,$);
+#6= IFCAPPLICATION(#5,'2024','GREY','REVOLT');
+
+#10= IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.0E-05,#11,$);
+#11= IFCAXIS2PLACEMENT3D(#12,$,$);
+#12= IFCCARTESIANPOINT((0.,0.,0.));
+
+#15= IFCUNITASSIGNMENT((#16));
+#16= IFCSIUNIT(*,.LENGTHUNIT.,.MILLI.,.METRE.);
+
+
+/* SITES AGGREGATED TO PROJECT */
+#29=IFCRELAGGREGATES('2te7nJfPvJyvPEahDDrZW0',#2,'ProjectContainer',$,#1,(#29990,#30000,#30010,#30020,#30030,#30040));
+#31=IFCLOCALPLACEMENT($,#11);
+
+ 
+#29990=IFCSITE('14oxcl5LHMZQ6iCWJKpe5w',#2,'SITE_withCorrectGuid',$,$,#31,$,$,.COMPLEX.,$,$,$,$,$);
+#30000=IFCSITE(30000,#2,'SITE_withIncorrectGuid','GUID is not a string',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 
+#30010=IFCSITE('14oxcl5',#2,'SITE_withIncorrectGuid','Invalid Length',$,#31,$,$,.ELEMENT.,$,$,$,$,$); 
+#30020=IFCSITE('1-+=cl5LHMZQ6iCWJKpe5w',#2,'SITE_withIncorrectGuid','Invalid Characters',$,#31,$,$,.PARTIAL.,$,$,$,$,$); 
+#30030=IFCSITE('b4oxcl5LHMZQ6iCWJKpe5w',#2,'SITE_withIncorrectGuid','Invalid Starting Character',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 
+#30040=IFCSITE('14oxcl5LHMZQ6iCWJKpe5w',#2,'SITE_withIncorrectGuid','Duplicated GUID',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 

```

Comment:
```suggestion
#30040=IFCSITE('04qxcl5LHMZQ7iCWJKpe5w',#2,'SITE_withIncorrectGuid','Duplicated GUID',$,#31,$,$,.COMPLEX.,$,$,$,$,$); 

```

`GlobalId` needs to be unique within a model to avoid schema errors.

Reply:
See answer above